pkg/cmd/release-payload-controller/controller.go

```diff
@@ -0,0 +1,127 @@
+package release_payload_controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/library-go/pkg/operator/events"
+	releasepayloadclient "github.com/openshift/release-controller/pkg/client/clientset/versioned/typed/release/v1alpha1"
+	releasepayloadinformer "github.com/openshift/release-controller/pkg/client/informers/externalversions/release/v1alpha1"
+	releasepayloadlister "github.com/openshift/release-controller/pkg/client/listers/release/v1alpha1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"time"
+)
+
+type Controller interface {
+	sync(ctx context.Context, key string) error
+}
+
+type ReleasePayloadController struct {
+	name string
+
+	releasePayloadNamespace string
+	releasePayloadLister    releasepayloadlister.ReleasePayloadLister
+	releasePayloadClient    releasepayloadclient.ReleaseV1alpha1Interface
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+
+	syncFn func(ctx context.Context, key string) error
+}
+
+func NewReleasePayloadController(
+	name string,
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	eventRecorder events.Recorder,
+	queue workqueue.RateLimitingInterface,
+	addDefaultEventHandler bool) *ReleasePayloadController {
+	c := &ReleasePayloadController{
+		name:                    name,
+		releasePayloadNamespace: releasePayloadNamespace,
+		releasePayloadLister:    releasePayloadInformer.Lister(),
+		releasePayloadClient:    releasePayloadClient,
+		eventRecorder:           eventRecorder,
+		queue:                   queue,
+	}
+
+	c.cachesToSync = append(c.cachesToSync, releasePayloadInformer.Informer().HasSynced)
+
+	if addDefaultEventHandler {
+		releasePayloadInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: c.Enqueue,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				c.Enqueue(newObj)
+			},
+			DeleteFunc: c.Enqueue,
+		})
+	}
+
+	return c
+}
+
+func (c *ReleasePayloadController) Enqueue(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid queue key '%v': %v", obj, err))
+		return
+	}
+	c.queue.Add(key)
+}
+
+func (c *ReleasePayloadController) Run(ctx context.Context) {
+	c.RunWorkers(ctx, 1)
+}
+
+func (c *ReleasePayloadController) RunWorkers(ctx context.Context, workers int) {
```

Comment:
this is your Run method.  Remove the other one.

Reply:
Done (in another, upcoming, PR)